eos-gpt2mbr

```diff
@@ -0,0 +1,159 @@
+if [ "$root_guid" != "4F68BCE3-E8CD-4DB1-96E7-FBCAF984B709" ]; then
+	echo "eos-gpt2mbr: $root_part doesn't have the correct GUID (read $root_guid)"
+	exit 1
+fi
+
+# check the last partition on the disk
+lastpart=$(echo "$parts" | sed -n -e '$ s/[^:]*\([0-9]\) :.*$/\1/p')
+
+if [ $lastpart -eq $swap_partno ]; then
+  # already have an extra partition, perhaps we were halfway through creating
+  # a swap partition but didn't finish. Remove it to try again below.
+  parts=$(echo "$parts" | sed '$d')
+elif [ $lastpart -gt $swap_partno ]; then
+  echo "repartition: found $lastpart partitions?"
+  exit 1
+fi
+
+# get the ESP
+esp_partno=$(echo "$parts" | sed -n -e "s/[^:]*\([0-9]\) :.*C12A7328-F81F-11D2-BA4B-00A0C93EC93B.*/\1/p")
+
+# check if ESP does exist
+if [ -z "${esp_partno}" ]; then
+	echo "eos-gpt2mbr: ESP not found"
+	exit 1
+fi
+
+esp_part=${root_disk}${esp_partno}
+esp_part_dev=${esp_part#/dev/}
+
+# remove ESP
+parts=$(echo "$parts" | sed -e "/$esp_part_dev/d")
+
+# calculate new partition sizes
+disk_size=$(blockdev --getsz $root_disk)
+part_size=$(blockdev --getsz $root_part)
+part_start=$(echo "$parts" | sed -n -e '$ s/.*start=[ ]\+\([0-9]\+\).*$/\1/p')
+part_end=$(( part_start + part_size ))
+echo "Dsize $disk_size Psize $part_size Pstart $part_start Pend $part_end"
+
+# Calculate the new root partition size, assuming that it will expand to fill
+# the remainder of the disk
+new_size=$(( disk_size - part_start ))
+
+# If we find ourselves with >100GB free space, we'll use the final 4GB as
+# a swap partition
+added_space=$(( new_size - part_size ))
+if [ $added_space -gt 209715200 ]; then
+  new_size=$(( new_size - 8388608 ))
+  swap_start=$(( part_start + new_size ))
+
+  # Align swap partition start to 1MB boundary
+  residue=$(( swap_start % 2048 ))
+  if [ $residue -gt 0 ]; then
+    swap_start=$(( swap_start + 2048 - residue ))
+    # might as well bump up root partition size too, instead of leaving a gap
+    new_size=$(( new_size + 2048 - residue ))
+  fi
+fi
+
+if [ $new_size -gt $part_size ]; then
+  echo "Try to resize $root_part to fill $new_size sectors"
+  parts=$(echo "$parts" | sed -e "$ s/size=[0-9\t ]*,/size=$new_size,/")
+fi
+
+if [ -n "$swap_start" ]; then
+  # Create swap partition
+  echo "Create swap partition at $swap_start"
+  parts="$parts
+start=$swap_start"
+fi
+
+# Reset partition indexes
+root_disk_dev=${root_disk#/dev/}
+parts=$(echo "$parts" | sed -e "s/${root_disk_dev}.//")
+
+# GPT -> DOS
+parts=$(echo "$parts" | sed -e "s/label: gpt/label: dos/")
+
+# Remove GPT types and UUIDs
+parts=$(echo "$parts" | sed -e "s/, type=.*//")
+
+echo "$parts"
+echo "$parts" | sfdisk --force --no-reread $root_disk
+udevadm settle
+
+# mkswap with the new partition index
+swap_partno=$((swap_partno - 1))
+swap_part=${root_disk}${swap_partno}
+[ -e "$swap_part" ] && mkswap -L eos-swap $swap_part
+
+# Remove GPT primary partition table
+dd if=/dev/zero of=${root_disk} conv=fsync bs=512 count=1 seek=1
```

Comment:
need to do a "udevadm settle" after each time the disk is opened for writing then closed, because udev will recreate the devices, there is a moment when they don't exist